source/data_model/encoded_data/EncodedData.py

```diff
@@ -34,7 +34,8 @@ def __init__(self, examples, labels: dict = None, example_ids: list = None, feat
                     .format(len(label), len(example_ids))
                 assert examples is None or len(example_ids) == examples.shape[0], "EncodedData: there are {} example ids, but {} examples."\
                     .format(len(example_ids), examples.shape[0])
-        assert len(labels.keys()) > 0 if labels is not None else True
+        if encoding not in ["MatchedReceptorsEncoder", "MatchedSequencesEncoder", "MatchedRegexEncoder"]:
```

Comment:
what is happening here? :) what is different with these encoders?

Reply:
These encoders are not used for machine learning. It is annoying to have to add fake labels to a dataset without labels just to run an encoding & report on it that do not use those labels

Reply:
Actually thinking about it again, if someone uses ExploratoryAnalysis labels should be entirely optional. Someone could also want to get out data encoded in a different way without having labels in the dataset